internal/gcsx/random_reader.go

```diff
@@ -211,19 +214,18 @@ func (rr *randomReader) tryReadingFromFileCache(ctx context.Context,
 
 		cacheHit = false
 		n = 0
+
 		if util.IsCacheHandleInvalid(err) {
-			err = rr.fileCacheHandle.Close()
-			if err != nil {
-				err = fmt.Errorf("tryReadingFromFileCache: while closing the fileCacheHandle: %v", err)
-				return
-			}
+			logger.Tracef("Closing cacheHandle:%p for object: %s//:%s", rr.fileCacheHandle, rr.bucket.Name(), rr.object.Name)
+			rr.fileCacheHandle.Close()
 			rr.fileCacheHandle = nil
 		} else if !strings.Contains(err.Error(), util.FallbackToGCSErrMsg) {
 			err = fmt.Errorf("tryReadingFromFileCache: while reading via cache: %v", err)
 			return
 		}
 		err = nil
 	}
+	logger.Tracef("%.13v -> ReadFromCache(%s, offset: %d, size: %d): %t", requestId, rr.object.Name, offset, len(p), cacheHit)
```

Comment:
should we use this in a defer statement and report the duration as well ? that way it will become consistent with GCS calls